### ISSUE TITLE
UseStrongestAuth "relogin" problematic on remote environments

### ISSUE THREAD

**Issue by codingllama** [gravitational/teleport#12675]
Expected behavior: tsh asks for (or allows) OTP MFA.

Current behavior:

Relogin attempts default to the cluster's strongest auth method since #11781 - this is an intentional workaround for stdin-stealing issues (#11709).

The issue arises when relogin is triggered in remote/containerized environments (tsh-on-remote, WSL2) - in those scenarios USB-based MFA is not an option. The current mitigation is to cancel the relogin attempt and explicitly run `tsh login`, which does allow for multiple authn methods.

See comment https://github.com/gravitational/teleport/issues/6630#issuecomment-1127807521.

Bug details:
- Teleport version: 9.2.1 (more precisely, any version after #11781)
- Recreation steps: see above.
- Debug logs: not necessary.

**Comment by gclawes** [comment]
This is probably a concern in enterprise SSO environments with the browser redirect to `127.0.0.1`.
```
$ tsh --proxy=teleport.car.nasdaq.com login
If browser window does not open automatically, open it by clicking on the link:
 http://127.0.0.1:55107/1ec1e096-8b55-4de1-8d4f-6994de6d3233
```

Doing either the `tsh login` above or a similar re-login via `kubectl` exec assumes you are running on your local workstation, similar to the assumption that you have access to USB for WEBAUTN.  I think in that case the only option is to forward the socket either with `ssh -L` or some kind of WSL2 socket forwarding (event then, the localhost socket isn't known in advance).

**Comment by codingllama** [comment]
From #13021:

I've put together a fix that should land in master shortly (I'll port it to v9 after it lands).

It's similar to the fix #11709, but this time I've gone ahead and made anything that isn't `tsh login` default to the strongest MFA method. If that doesn't work for you (eg, remote machine), then you can use the `--mfa-mode` flag to specify a different MFA method (eg, `tsh --mfa-mode=otp ssh llama@themachine`). Setting the `TELEPORT_MFA_MODE` environment variable also works.

I'll keep the present issue (#12675) focused on MFA only, as SSO is a different issue altogether.
